internal/controller/bucket/create.go

```diff
@@ -50,8 +50,8 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 		return managed.ExternalCreation{}, errors.New(errMissingS3Backend)
 	}
 
-	updated := atomic.Bool{}
-	errorsLeft := 0
+	atomicBool := atomic.Bool{}
```

Comment:
IMO: The name atomicBool doesn't explain its responsibility.
I think a name like `bucketCreated` is better (it's a little bit long, though)

And putting some comments is better for understanding in the future:
```go
// This value shows a bucket on one backend is already created.
// It is used to prevent goroutines to send multiple messages to `readyChan`.
```

Reply:
Good idea, added.